File: JavaArchitecture/app/src/main/java/com/demo/architecture/injection/component/ConfigPersistentComponent.java
package com.demo.architecture.injection.component;


import com.demo.architecture.injection.ConfigPersistent;
import com.demo.architecture.injection.module.ActivityModule;
import com.demo.architecture.injection.module.FragmentModule;

import dagger.Component;

/**
 * Created by HeYingXin on 2018/4/17.
 */
@ConfigPersistent
@Component(dependencies = AppComponent.class)
public interface ConfigPersistentComponent {
    ActivityComponent activityComponent(ActivityModule activityModule);
    FragmentComponent fragmentComponent(FragmentModule fragmentModule);
}

File: kotlin-frist-branch-TODO/app/src/main/java/com/mazouri/mvpkotlin/injection/component/ConfigPersistentComponent.kt
package com.mazouri.mvpkotlin.injection.component

import com.mazouri.mvpkotlin.injection.ConfigPersistent
import com.mazouri.mvpkotlin.injection.module.ActivityModule
import com.mazouri.mvpkotlin.injection.module.FragmentModule
import dagger.Component

/**
 * Created by HeYingXin on 2018/4/17.
 */
@ConfigPersistent
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface ConfigPersistentComponent {

    fun activityComponent(activityModule: ActivityModule): ActivityComponent

    fun fragmentComponent(fragmentModule: FragmentModule): FragmentComponent

}